### PR TITLE
Change temporary variable owner to always be enclosing class

### PR DESCRIPTION
**Pull request by Nargeshdb**

### CODE REVIEW

File: object-construction-checker/tests/mustcall/CrashOnCommon.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import java.net.*;
+
+class CrashOnCommon {
+//    Object baz = new Object(new Socket());
```
Comment (msridhar) [inline object-construction-checker/tests/mustcall/CrashOnCommon.java:4]: Can we add the field declaration test as well and make sure we get an error?

File: must-call-checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java [hunk 2]
```diff
@@ -67,8 +67,8 @@ public MustCallAnnotatedTypeFactory(final BaseTypeChecker checker) {
     TOP = AnnotationBuilder.fromClass(elements, MustCallUnknown.class);
     BOTTOM = createMustCall();
     POLY = AnnotationBuilder.fromClass(elements, PolyMustCall.class);
-    addAliasedAnnotation(InheritableMustCall.class, MustCall.class, true);
-    addAliasedAnnotation(MustCallChoice.class, POLY);
+    addAliasedTypeAnnotation(InheritableMustCall.class, MustCall.class, true);
```
Comment (msridhar) [inline must-call-checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java:70]: Let's not include this unrelated change.  Probably @kelloggm has fixed on his branch

File: must-call-checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java [hunk 3]
```diff
@@ -204,7 +204,7 @@ private void changeNonOwningParametersTypes(
   public AnnotatedTypeMirror fromElement(Element elt) {
     AnnotatedTypeMirror type = super.fromElement(elt);
     // Support @InheritableMustCall meaning @MustCall on all class declaration elements.
-    if (ElementUtils.isClassElement(elt)) {
+    if (ElementUtils.isTypeElement(elt)) {
```
Comment (msridhar) [inline must-call-checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java:207]: Let's not include this unrelated change. Probably @kelloggm has fixed on his branch

**Review by msridhar — CHANGES_REQUESTED** [review]

File: object-construction-checker/tests/mustcall/CrashOnCommon.java (hunk 1 above)
Comment (Nargeshdb) [reply object-construction-checker/tests/mustcall/CrashOnCommon.java]: https://github.com/kelloggm/object-construction-checker/pull/300/commits/a71831579d604bf70bc86fe342299488b4f80141

File: must-call-checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java (hunk 2 above)
Comment (Nargeshdb) [reply must-call-checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java]: https://github.com/kelloggm/object-construction-checker/pull/300/commits/a71831579d604bf70bc86fe342299488b4f80141

**Review by msridhar — APPROVED** [review]
